### PR TITLE
Adding Dismiss Keyboard on Tap

### LINKED ISSUE #28: Bug: Keyboard Tap to dismiss
**Describe the bug**
in `TaskDetailsViewController` once the keyboard is up it does not dismiss under any circumstance.

**To Reproduce**
1. Go to Tasks list.
2. Click on `+` button.
3. Tap on `TaskTextField`
4. Now the keyboard cannot be dismissed

**Expected behavior**
The keyboard should dismiss when tapped anywhere on the view or when return key is pressed.

**Screenshots**
N/A

**Testing (please complete the following information):**
 - Device: iPhone 11
 - iOS: 14.0
 - Xcode 11.7

**Additional context**
you can refer [here](https://medium.com/@KaushElsewhere/how-to-dismiss-keyboard-in-a-view-controller-of-ios-3b1bfe973ad1) for all possible ways to solve this issue.

### PR DESCRIPTION
**Pull request by Monkeyank**
### Description
Added the functionality to dismiss the keyboard on tapping outside the keyboard.

Fixes #28 

### Type of Change:
**Delete irrelevant options.**

- Code
- User Interface

### How Has This Been Tested?
I tested it on a simulator by clicking the + button on the upper right corner. I then tested each of the text fields.

### Checklist:
**Delete irrelevant options.**

- [x] My PR follows the style guidelines of this project
- [x] I have performed a self-review of my own code or materials
- [x] I have commented my code or provided relevant documentation, particularly in hard-to-understand areas

**Code/Quality Assurance Only**
- [x] My changes generate no new warnings 
- [x] My PR currently breaks something (fix or feature that would cause existing functionality to not work as expected)

### CODE REVIEW

**Comment by aaryankotharii** [comment]
Hi @Monkeyank , firstly thank you so much for your contribution. This solution is perfect for the issue described in #28, Apart from hiding the keyboard from tapping the view we can also hide it when the user taps return key of the keyboard. How does that sound?

**Comment by Monkeyank** [comment]
> Hi @Monkeyank , firstly thank you so much for your contribution. This solution is perfect for the issue described in #28, Apart from hiding the keyboard from tapping the view we can also hide it when the user taps return key of the keyboard. How does that sound?

Sounds Good! That should be an easy tweak!

**Comment by aaryankotharii** [comment]
> Sounds Good! That should be an easy tweak!

Great! you can re-request a review once done.

**Review by aaryankotharii — CHANGES_REQUESTED** [review]
dismiss keyboard with return key.

**Comment by Monkeyank** [comment]
@aaryankotharii I added dismissKeyboard on return. I didn't add it to the SubTask body, because when you return, you should be able go to the next line. When taking notes you want to be able to go to the next line without closing the keyboard every time.

File: Application/To-Do/Controller/TaskDetailsViewController.swift [hunk 1]
```diff
@@ -80,3 +84,14 @@ class TaskDetailsViewController: UIViewController {
     }
     
 }
+extension TaskDetailsViewController {
+func textFieldShouldReturn(_ textField: UITextField) -> Bool {
+//Check if there is any other text-field in the view whose tag is +1 greater than the current text-field on which the return key was pressed. If yes → then move the cursor to that next text-field. If No → Dismiss the keyboard
```
Comment (aaryankotharii) [inline Application/To-Do/Controller/TaskDetailsViewController.swift:89]: @Monkeyank I must say I am slightly confused with the approach you have taken. Great job with identifying that keyboard should not dismiss while return is clicked on textview. That said, how about we add a single if condition for `taskTextField` and dismiss keyboard in that.
you code should  look something like this
``` Swift
func textFieldShouldReturn(_ textField: UITextField) -> Bool {
    if textField == taskTitleTextField{
        textField.resignFirstResponder()
        return true
    }
    return false
    }
```
Please check this out and let me know if this works the same as the solution provided by you.

**Review by aaryankotharii — CHANGES_REQUESTED** [review]

File: Application/To-Do/Controller/TaskDetailsViewController.swift [hunk 2]
```diff
@@ -13,7 +13,7 @@ protocol TaskDelegate: class {
     func didTapUpdate(task : Task)
 }
 
-class TaskDetailsViewController: UIViewController {
+class TaskDetailsViewController: UIViewController, UITextFieldDelegate, UITextViewDelegate {
```
Comment (aaryankotharii) [inline Application/To-Do/Controller/TaskDetailsViewController.swift:16]: Also please conform the delegates to the extension where its methods are to make the purpose of the extension clearer.

**Comment by Monkeyank** [comment]
@aaryankotharii Made the changes requested

**Comment by aaryankotharii** [comment]
@Monkeyank I see that there are merge conflicts in `TaskDetailsViewController`. Could you please fetch the latest changes from the upstream master.

**Comment by Monkeyank** [comment]
@aaryankotharii done!

**Comment by aaryankotharii** [comment]
Great @Monkeyank, Thank you for your contribution!
Approved.

**Review by aaryankotharii — APPROVED** [review]

**Pull request merged**